### PR TITLE
API: Get release by tags endpoint

### PR DESCRIPTION
**Pull request by johanvdw**
Get a release based on a tag name (for which a release exists).
Based on:
https://developer.github.com/v3/repos/releases/#get-a-release-by-tag-name

This is a useful feature for deploy scripts. 

Note I'm new to gitea and go in general, so please review carefully.

### CODE REVIEW

File: routers/api/v1/repo/release_tags.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	//   required: true
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/Release"
+
+	tag := ctx.Params(":tag")
+
+	release, err := models.GetRelease(ctx.Repo.Repository.ID, tag)
+	if err != nil {
+		ctx.Error(http.StatusNotFound, "GetRelease", err)
```
Comment (6543) [inline routers/api/v1/repo/release_tags.go]:
```suggestion
		if models.IsErrReleaseNotExist(err) {
				ctx.Error(http.StatusNotFound, "GetRelease", err)
				return
		}
		ctx.Error(http.StatusInternalServerError, "GetRelease", err)
```

File: routers/api/v1/repo/release_tags.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	//   type: string
+	//   required: true
+	// - name: tag
+	//   in: path
+	//   description: tagname of the release to get
+	//   type: string
+	//   required: true
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/Release"
```
Comment (a1012112796) [inline routers/api/v1/repo/release_tags.go:39]:
```suggestion
	//   "200":
	//     "$ref": "#/responses/Release"
	//   "404":
	//     "$ref": "#/responses/notFound"
```

File: routers/api/v1/repo/release_tags.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
+)
+
+// GetReleaseTag get a single release of a repository by its tagname
+func GetReleaseTag(ctx *context.APIContext) {
+	// swagger:operation GET /repos/{owner}/{repo}/releases/tags/{tag} repository repoGetReleaseTag
+	// ---
+	// summary: Get a release by tagname
```
Comment (a1012112796) [inline routers/api/v1/repo/release_tags.go]:
```suggestion
	// summary: Get a release by tag name
```

File: routers/api/v1/repo/release_tags.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	if err != nil {
+		ctx.Error(http.StatusNotFound, "GetRelease", err)
+		return
+	}
+
+	if err := release.LoadAttributes(); err != nil {
+		ctx.Error(http.StatusInternalServerError, "LoadAttributes", err)
+		return
+	}
+	ctx.JSON(http.StatusOK, release.APIFormat())
```
Comment (a1012112796) [inline routers/api/v1/repo/release_tags.go:59]: @6543 This function also call ``user.APIFormat()`` to render publisher message, has any other way? Thanks

Comment (6543) [reply routers/api/v1/repo/release_tags.go:59]: @a1012112796 need refactor - out of scope of this pull :+1:

**Comment by 6543** [comment]
some tests missing
-> https://github.com/go-gitea/gitea/blob/master/integrations/api_releases_test.go

just add 2 api calls: (Get Existing Release, Get Not Existing Release) 
and check result :)

**Comment by johanvdw** [comment]
> some tests missing
> -> https://github.com/go-gitea/gitea/blob/master/integrations/api_releases_test.go
> 
> just add 2 api calls: (Get Existing Release, Get Not Existing Release)
> and check result :)
> 
> Existing Releases it tests: https://github.com/go-gitea/gitea/blob/master/models/fixtures/release.yml

Thanks for the pointer, I was looking for the tests. I will add them.

**Comment by 6543** [comment]
You need to commit result of `make generate-swagger` again :)

**Review by 6543 — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
@johanvdw It's feature freeze, so we have to wait for 1.13.0 RC1

**Comment by zeripath** [comment]
meh if it's ready - it's ready. I'm happy to merge into 1.13 given it's self-contained.

**Pull request merged**